company-clang.el

```diff
@@ -109,8 +179,9 @@ or automatically through a custom `company-clang-prefix-guesser'."
 
 ;; TODO: Handle Pattern (syntactic hints would be neat).
 ;; Do we ever see OVERLOAD (or OVERRIDE)?
+;; Parse completion's comments/documentation too
 (defconst company-clang--completion-pattern
-  "^COMPLETION: \\_<\\(%s[a-zA-Z0-9_:]*\\)\\(?: : \\(.*\\)$\\)?$")
+   "^COMPLETION: \\_<\\(%s[a-zA-Z0-9_:]*\\) : \\(.*?\\)\\(?: : \\(.*\\)\\)\\{,1\\}$")
```

Comment:
I wrote [regexp-testing-ground.el] (https://github.com/mehw/company-mode/blob/clang-doc-testing/regexp-testing-ground.el), you may add/change *prefixes*, *regexps*, *strings*, it should give clear answers: call `regexp-testing-ground-run-tests` and look at the buffer `*regexp-testing-ground*` for the results (add the file's load path and `(require 'regexp-testing-ground)` to `~/.emacs`:
* `"^COMPLETION: \_<\(%s[a-zA-Z0-9_:]*\)\(?: : \(.+?\)?\)?\(?: : \(.*\)?\)?$"`: **wrong result**: "COMPLETION: foobar : [#int#]foobar(<#int a#>) : " group 3 is an empty string instead of nil.
* `"^COMPLETION: \_<\(%s[a-zA-Z0-9_:]*\)\(?: : \(.+?\)?\)?\(?: : \(.+?\)?\)?$"` and `"^COMPLETION: \_<\(%s[a-zA-Z0-9_:]*\)\(?: : \(.+?\)?\)?\(?: : \(.+\)?\)?$"` give **same results**.

The `+` mean *at least one match*, while `*` means *no match or any match*.
```lisp
prefix : 'foo'
regexp : '^COMPLETION: \_<\(%s[a-zA-Z0-9_:]*\)\(?: : \(.+?\)?\)?\(?: : \(.+?\)?\)?$'
 pattern: '^COMPLETION: \_<\(foo[a-zA-Z0-9_:]*\)\(?: : \(.+?\)?\)?\(?: : \(.+?\)?\)?$'
  string : 'COMPLETION: '
   group 1: nil
   group 2: nil
   group 3: nil
  string : 'COMPLETION: blabla'
   group 1: nil
   group 2: nil
   group 3: nil
  string : 'COMPLETION: foobar'
   group 1: 'foobar'
   group 2: nil
   group 3: nil
  string : 'COMPLETION: foobar : '
   group 1: 'foobar'
   group 2: nil
   group 3: nil
  string : 'COMPLETION: foobar : [#int#]foobar(<#int a#>)'
   group 1: 'foobar'
   group 2: '[#int#]foobar(<#int a#>)'
   group 3: nil
  string : 'COMPLETION: foobar : [#int#]foobar(<#int a#>) : '
   group 1: 'foobar'
   group 2: '[#int#]foobar(<#int a#>)'
   group 3: nil
  string : 'COMPLETION: foobar : [#int#]foobar(<#int a#>) : Comment 1'
   group 1: 'foobar'
   group 2: '[#int#]foobar(<#int a#>)'
   group 3: 'Comment 1'
  string : 'COMPLETION: foobar : [#int#]foobar(<#int a#>) : Comment 1 : '
   group 1: 'foobar'
   group 2: '[#int#]foobar(<#int a#>)'
   group 3: 'Comment 1 : '
```